roffit

```diff
@@ -256,6 +256,33 @@ sub linkfile {
             # convert https://, http:// and ftp:// URLs to <a href> links
             $field =~ s/(^|\W)((https|http|ftp):\/\/[a-z0-9\-._~%:\/?\#\[\]\@!\$&'()*+,;=]+)/$1<a href=\"$2\">$2<\/a>/gi;
 
+            # remove punctuation characters at end of url
+            $field =~ s/(href=\")([^\"]*)(&quot;)/$1$2/g;
+            $field =~ s/(href=\"(http:\/\/|https:\/\/|ftp:\/\/)\")*([,.;()*])*(\">)/$1$4/g;
+
+            # null empty protocols
+            $field =~ s/href=\"(http:\/\/|https:\/\/|ftp:\/\/)\"/href=\"javascript:void(0)\"/g;
```

Comment:
If the above is fixed, then this kind of link should not get generated. Right? And I would prefer not having this a link at all rather than to have this javascript-thing. In fact, roffit generates HTML (and CSS) only, I don't want to add javascript calls to the mix.